packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -34,6 +34,223 @@ export interface HttpRouteMatch {
    * and you want the route to match requests to my-service.local/metrics, your prefix should be /metrics.
    */
   readonly prefixPath: string;
+
+  /**
+   * Specifies the client request headers to match on. All specified headers
+   * must match for the route to match.
+   *
+   * @default - do not match on headers
+   */
+  readonly headers?: HeaderMatch[];
+
+  /**
+   * The client request method to match on. Specify only one.
+   *
+   * @default - do not match on request method
+   */
+  readonly method?: HttpRouteMatchMethod;
+
+  /**
+   * The client request scheme to match on. Specify only one. Applicable only for HTTP2 routes.
+   *
+   * @default - do not match on HTTP2 scheme
+   */
+  readonly scheme?: HttpRouteMatchScheme;
+}
+
+/**
+ * Supported request matching methods.
+ */
+export enum HttpRouteMatchMethod {
+  /**
+   * GET request
+   */
+  GET = 'GET',
+
+  /**
+   * HEAD request
+   */
+  HEAD = 'HEAD',
+
+  /**
+   * POST request
+   */
+  POST = 'POST',
+
+  /**
+   * PUT request
+   */
+  PUT = 'PUT',
+
+  /**
+   * DELETE request
+   */
+  DELETE = 'DELETE',
+
+  /**
+   * CONNECT request
+   */
+  CONNECT = 'CONNECT',
+
+  /**
+   * OPTIONS request
+   */
+  OPTIONS = 'OPTIONS',
+
+  /**
+   * TRACE request
+   */
+  TRACE = 'TRACE',
+
+  /**
+   * PATCH request
+   */
+  PATCH = 'PATCH',
+}
+
+/**
+ * Supported :scheme options for HTTP2
+ */
+export enum HttpRouteMatchScheme {
+  /**
+   * Match HTTP requests
+   */
+  HTTP = 'http',
+
+  /**
+   * Match HTTPS requests
+   */
+  HTTPS = 'https',
+}
+
+/**
+ * Configuration for `HeaderMatch`
+ */
+export interface HeaderMatchConfig {
+  /**
+   * The Http header name
+   */
+  readonly name: string;
+
+  /**
+   * Invert the matching condition.
+   *
+   * @default false
+   */
+  readonly invert?: boolean;
+
+  /**
+   * The match property.
+   */
+  readonly matchProperty: CfnRoute.HeaderMatchMethodProperty;
+}
+
+/**
+ * Used to generate header matching methods.
+ */
+export abstract class HeaderMatch {
+  /**
+   * The value sent by the client must match the specified value exactly.
```

Comment:
This "The value sent by the client" is a weird phrasing I feel like.

How about "The value of the header with the given name in the request" instead?